### ISSUE TITLE
Switch orientation run time when recording stream

### ISSUE THREAD

**Issue by ravigautam7786** [jgh-/VideoCore#192]
Hi,

Please help me.

I want to record video in both portrait and landscape mode when user move device from portrait to landscape. But currently video is not going in landscape mode.

Thanks

**Comment by jgh-** [comment]
You would have to restart the session to switch the actual video between landscape and portrait.

**Comment by ravigautam7786** [comment]
Thanks,

I will try the same and let you know the updates.

**Comment by ravigautam7786** [comment]
Please let me know how I will restart the session. I tried with [_session endRtmpSession]; and 
   [_session startRtmpSessionWithURL:RTMPURL andStreamKey:_strStreamName]; with new video size but in this technique live stream is stopped on the web player and I refreshed the page again then it was reflected.

Thanks

**Comment by ravigautam7786** [comment]
And taking memory while restarting session.

**Comment by aramikg** [comment]
Not sure I actually ran into the same issue... just noticed that videoSize is only update on start session... let me know if you find a way.

**Comment by ravigautam7786** [comment]
James,
Please let me know the updates on this.
Thanks

**Comment by maxcampolo** [comment]
@ravigautam7786 This comment applies to changing resolution but it may help in your case. I found when I make a change to the video size I have to destroy the session and recreate it, rather than just starting and stopping it. Here is some sample code from a method I use to do this.

```
- (void)restartSessionWithCompletion:(void(^)(void))completion {
 // If there is a session reset it
    if (_session) {
        [_session.previewView removeFromSuperview];
        _session = nil;
        _session.delegate = nil;
    }
    
// Initiate new session
    _session = [[VCSimpleSession alloc] initWithVideoSize:[self getStoredResolution] frameRate:30 bitrate:[self getStoredBitrate] useInterfaceOrientation:YES cameraState:VCCameraStateBack];

  
 // Add the camera preview of session as a subview again
    [self.previewView insertSubview:_session.previewView atIndex:0];
    _session.previewView.frame = self.previewView.bounds;
    _session.delegate = self;

// I use this completion handler to start the session again after all these UI updates have completed
    if (completion != nil) {
        completion();
    }
}
```
Hope this helps. @jgh- if this is useful I can add something like it to the example in the repo.

**Comment by ravigautam7786** [comment]
maxcampolo thanks for reply.

I will try the same and will provide the updates.

**Comment by ravigautam7786** [comment]
Tested this code but problem is remain same.
When i am restarting session again with this code my video stopped on other player and i have to restart it.
And my video is not recorded on server accordingly.
Please give me any suggestion.

Thanks
Ravi Gautam

**Comment by ravigautam7786** [comment]
James
Can you help me on this
Thanks

**Comment by ravigautam7786** [comment]
James
This is not woking.Please give me a way

Thanks
Ravi Gautam

**Comment by ravigautam7786** [comment]
Is anyone can help me on this?

**Comment by ravigautam7786** [comment]
James,
I am stuck very badly with this. Please give me any suggestion.
Is there any way to rotate frames run time without interrupting current session. Because all these way are closing my existing session.

Thanks
Ravi Gautam

**Comment by JALsnipe** [comment]
Stop spamming.  If someone found a solution, they would have posted.

**Comment by jgh-** [comment]
Yeah, I think there is probably a bug in the VCSimpleSession code, but what you can do is deallocate the session and then reinitialize/start it with your new settings

**Comment by ravigautam7786** [comment]
But when I deallocate the current session my live streaming is closed on other end at that time and I have to restart it on other end to play again.

**Comment by jgh-** [comment]
This is a shortcoming of the RTMP protocol.  As far as I know there's no way of dynamically changing the dimensions of a video during a stream.

**Comment by ravigautam7786** [comment]
So I can do anything for it

Thanks
Ravi Gautam
